### DISCUSSION TITLE
The number of output solution set is not equals to pop size

### DISCUSSION THREAD

**Discussion by superyue72** [anyoptimization/pymoo#568] [Q&A]
Hi there

I'm using the NSGA2, RNSGA2, and SMSEMOA from Pymoo.

I set the pop size as 100, but the solution generated by the algorithm contains less than 100 solutions. Sometimes even less than 10.

For example, as shown in the figure, I got 23 solutions, but I set the pop size to 100.

Does anyone know why this would happen? How could I revise it?

Many thanks!

<img width="547" alt="image" src="https://github.com/anyoptimization/pymoo/assets/109505926/c527e24c-30b4-45a1-9444-4e7f71967dfe">

**Comment by blankjul** [comment]
Yes the algorithm only returns the non-dominated set of solutions. You can get the `res.pop` object to get the whole population as well.
